src/Server/Services/Jobs/JobStore.cs

```diff
@@ -0,0 +1,297 @@
+using Polly;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nvidia.Clara.DicomAdapter.Server.Services.Jobs
+{
+    public class JobItemStatus
+    {
+        internal static readonly JobItemStatus Default = new JobItemStatus();
+    }
+
+    public class JobStore : IHostedService, IJobStore
+    {
+        private const int MaxRetryLimit = 3;
+        private static readonly object SyncRoot = new Object();
+
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IOptions<DicomAdapterConfiguration> _configuration;
+        private readonly ILogger<JobStore> _logger;
+        private readonly IKubernetesWrapper _kubernetesClient;
+        private readonly IFileSystem _fileSystem;
+        private CustomResourceWatcher<JobCustomResourceList, JobCustomResource> _watcher;
+        private readonly BlockingCollection<JobCustomResource> _jobs;
+        private readonly HashSet<string> _jobIds;
+
+        public JobStore(
+            ILoggerFactory loggerFactory,
+            ILogger<JobStore> logger,
+            IOptions<DicomAdapterConfiguration> configuration,
+            IKubernetesWrapper kubernetesClient,
+            IFileSystem fileSystem)
+        {
+            _loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _kubernetesClient = kubernetesClient ?? throw new ArgumentNullException(nameof(kubernetesClient));
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _jobs = new BlockingCollection<JobCustomResource>();
+            _jobIds = new HashSet<string>();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            var task = Task.Run(async () =>
+            {
+                await BackgroundProcessing(cancellationToken);
+            });
+
+            if (task.IsCompleted)
+                return task;
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.Log(LogLevel.Information, "Job Store Hosted Service is stopping.");
+            return Task.CompletedTask;
+        }
+
+        public async Task New(Job job, string jobName, IList<InstanceStorageInfo> instances)
+        {
+            Guard.Against.Null(job, nameof(job));
+            Guard.Against.Null(jobName, nameof(jobName));
+            Guard.Against.NullOrEmpty(instances, nameof(instances));
+
+            var inferenceJob = CreateInferenceJob(job, jobName, instances);
+            MakeACopyOfPayload(inferenceJob);
```

Comment:
Some comments on why copying is needed will help.

Reply:
Done.